### PR TITLE
[expo] Add types for SecureStore keychain item accessibility constants

### PR DESCRIPTION
**Pull request by jkillian**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.expo.io/versions/latest/sdk/securestore#exposecurestoresetitemasynckey-value-options
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/expo/expo-tests.tsx [hunk 1]
```diff
@@ -484,6 +485,19 @@ async () => {
     isString(queryParams2['y'] || '');
 };
 
+// #region securestore
+async () => {
+    await SecureStore.setItemAsync('some-key', 'some-val', {
+        keychainAccessible: SecureStore.WHEN_PASSCODE_SET_THIS_DEVICE_ONLY,
```
Comment (moshfeu) [inline types/expo/expo-tests.tsx]: Maybe you can test the `keychainService` option too.

File: types/expo/expo-tests.tsx [hunk 2]
```diff
@@ -484,6 +485,19 @@ async () => {
...
+    });
+    const result = await SecureStore.getItemAsync('some-key');
+    if (result != null) {
+        result.slice() === 'some-val';
+    }
+    await SecureStore.deleteItemAsync('some-key');
+};
```
Comment (moshfeu) [inline types/expo/expo-tests.tsx]: Other `keychainAccessible` options can be tested here. Something like this:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/expo/expo-tests.tsx#L160

File: types/expo/index.d.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@
 //                 Levan Basharuli <https://github.com/levansuper>
 //                 Pavel Ihm <https://github.com/ihmpavel>
 //                 Bartosz Dotryw <https://github.com/burtek>
+//                 Jason Killian <https://github.com/jkillian>
```
Comment (moshfeu) [inline types/expo/index.d.ts]: 👍

**Review by moshfeu — CHANGES_REQUESTED** [review]

**Review by moshfeu — APPROVED** [review]
Thanks!

**Comment by JKillian** [comment]
Glad to get this merged, thanks for the reviews @moshfeu!

**Pull request merged**